app/controllers/api/users_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Api::UsersController < ApplicationController
+  def show
+    @user = User.find_by(id: params[:id])
+    @message = 'Not Found'
+    if @user
+      render 'show', format: :json, status: :ok
+    else
+      render 'error', format: :json, status: :not_found
+    end
```

Comment:
```suggestion
    @user = User.find_by(id: params[:id])
    if @user
      render 'show', format: :json, status: :ok
    else
      @message = 'Not Found'
      render 'error', format: :json, status: :not_found
    end
```

if文の前で`@message`を定義してもUserが存在するときは使われないので`render 'error'`の直前で定義するのでいいと思います